airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py

```diff
@@ -404,131 +406,6 @@ def configure_request_params_mock(stream_1, stream_2):
     stream_2.request_params.return_value = {"q": "query"}
 
 
-def test_rate_limit_bulk(stream_config, stream_api, bulk_catalog, state):
```

Comment:
I'm not clear why these rate limit tests should be deleted.

Reply:
They are very hard to maintain as they rely on private things to run for example `page_size`. Also, the way they are instantiated makes then not have the ConcurrentCursor as it relies on aggressively re-implementing source methods (like `source.streams = Mock()`). We will move these to mock server tests. We have already added a test than ensure it retries on 406 so we know the integration with the error handling works so I'm not too worried in removing this test specifically.